src/test/java/teammates/test/cases/util/StringHelperTest.java

```diff
@@ -412,4 +412,13 @@ public void testExtractContentFromQuotedString() {
         assertEquals(StringHelper.extractContentFromQuotedString(null), null);
         assertEquals(StringHelper.extractContentFromQuotedString(""), "");
     }
+
+    @Test
+    public void testSignatureGenerationDeterministic() {
```

Comment:
In addition to verify determinism, we should verify if it is correct. (i.e. the returned value is indeed the expected hash of a certain String. You can take a look at the encrypt test methods.